drivers/sensor/tmp116/tmp116.c

```diff
@@ -36,31 +36,44 @@ static int tmp116_reg_read(const struct device *dev, uint8_t reg,
 	return 0;
 }
 
+static int tmp116_reg_write(const struct device *dev, uint8_t reg,
+			    uint16_t val)
+{
+	struct tmp116_data *drv_data = dev->data;
+	const struct tmp116_dev_config *cfg = dev->config;
+	uint8_t tx_buf[3] = {reg, val >> 8, val & 0xFF};
+
+	return i2c_write(drv_data->i2c, tx_buf, sizeof(tx_buf),
+			cfg->i2c_addr);
+}
+
 /**
  * @brief Check the Device ID
  *
  * @param[in]   dev     Pointer to the device structure
  *
- * @retval 0 On success
+ * @retval device id On success
```

Comment:
I think it's better to use an argument for output value and leave return code for errors only

Reply:
Used argument to return the ID and used return value for error code as you suggested.